src/Functions/FunctionsStringDistance.cpp

```diff
@@ -226,6 +226,34 @@ struct ByteJaccardIndexImpl
 
 static constexpr size_t max_string_size = 1u << 16;
 
+void parseUTF8String(PaddedPODArray<UInt32> & code_points, const char * __restrict data, size_t size)
```

Comment:
Looks very similar to parsing a string in `ByteJaccardIndexImpl`, do you think we can deduplicate it a little bit?